machine/machine_core/machine_virtual.py

```diff
@@ -360,8 +360,13 @@ def execute(*args):
         image_to_use = self.image_file
         if not self.maintain:
             (unused, self._transient_image) = tempfile.mkstemp(suffix='.qcow2', prefix="", dir=self.run_dir)
+            options = ""
+            # check if it is qcow2 (libvirt complains otherwise)
+            with open(self.image_file, "rb") as f:
+                if b"qcow2" in f.read(128):
```

Comment:
out of interest, is it given that first 128 chars always contains type of format?

Reply:
I tried it on a couple of qcow images that I had laying around, and it looks quite regular:
```
❱❱❱ od -A x -t x1z -v test/images/fedora-32 | head
000000 51 46 49 fb 00 00 00 03 00 00 00 00 00 00 01 48  >QFI............H<
000010 00 00 00 70 00 00 00 10 00 00 00 03 00 00 00 00  >...p............<
000020 00 00 00 00 00 00 00 18 00 00 00 00 00 03 00 00  >................<
000030 00 00 00 00 00 01 00 00 00 00 00 01 00 00 00 00  >................<
000040 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
000050 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
000060 00 00 00 04 00 00 00 68 e2 79 2a ca 00 00 00 05  >.......h.y*.....<
000070 71 63 6f 77 32 00 00 00 68 03 f8 57 00 00 00 c0  >qcow2...h..W....<
000080 00 00 64 69 72 74 79 20 62 69 74 00 00 00 00 00  >..dirty bit.....<
000090 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  >................<
```

Reply:
Actually, our cirros standard image doesn't have it -- let's rather check for `QFI` then.

Reply:
See https://people.gnome.org/~markmc/qcow-image-format.html, the `QFI` magic is fixed.